src/generate/templates/ping-device-ping1d.cpp.in

```diff
@@ -0,0 +1,91 @@
+#include <ping-device-ping1d.h>
+#include <ping-message-common.h>
+#include <ping-message-ping1d.h>
+
+Ping1d::~Ping1d()
+{
+{% for msg in messages["get"]|sort %}
+{% for field in messages["get"][msg].payload %}
+{% if generator.is_vector(field.type) %}
+  if ({{field.name}}) {
+   free({{field.name}});
+  }
+{% endif %}
+{% endfor %}
+{% endfor %}
+}
+
+bool Ping1d::initialize(uint16_t pingIntervalMs)
+{
+  if (!PingDevice::initialize()) {
+    return false;
+  }
+
+  // Configure ping interval
+  if (!set_ping_interval(pingIntervalMs)) {
+    return false;
+  }
+
+  return true;
+}
+
+void Ping1d::_handleMessage(const ping_message* message)
+{
+  switch (message->message_id()) {
+{% for msg in messages["get"]|sort %}
+{% if not messages["get"][msg].deprecated %}
+    case Ping1dId::{{msg|upper}}:
+    {
```

Comment:
The style of this project makes me pretty confuse, we are using 4 spaces in all repositories and this is a mix of 2 and 4 spaces, I believe that we should stay with 4 spaces as the others C++ and Python projects.

Reply:
this is going to be a pain in the ass to fix.

Reply:
done. I couldn't get vscode do this automatically :face_with_head_bandage: